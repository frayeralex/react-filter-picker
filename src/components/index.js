export { default as FilterItem } from './FilterItem/FilterItem';
export { default as FilterButton } from './FilterButton/FilterButton';
export { default as CollapsedBox } from './CollapsedBox/CollapsedBox';
export { default as FilterGroup } from './FilterGroup/FilterGroup';
export { default as FilterPicker } from './FilterPicker/FilterPicker';
