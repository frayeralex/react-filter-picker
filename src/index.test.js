import React from 'react';
import Filter from './index';
import { FilterButton, FilterPicker } from './components';

jest.mock('mobile-detect');

const props = {
  md: {
    phone: jest.fn(),
    tablet: jest.fn(),
  },
  document: {
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
  },
  filterData: {
    color: [
      {
        id: '1',
        title: 'first',
      },
      {
        id: '2',
        title: 'second',
      },
    ],
    size: [
      {
        id: '1',
        title: 'first',
      },
      {
        id: '2',
        title: 'second',
      },
    ],
  },
};

const coordinates = {
  top: 10,
  left: 10,
  right: 10,
  bottom: 30,
  width: 100,
  height: 40,
};

describe('<Filter/>', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<Filter {...props} />);
    expect(wrapper).toMatchSnapshot();
    expect(props.md.tablet).toHaveBeenCalled();
    expect(props.md.phone).toHaveBeenCalled();
    expect(props.document.addEventListener).not.toHaveBeenCalled();
    expect(props.document.removeEventListener).not.toHaveBeenCalled();
  });
  it('should render with menu include More Filter button', () => {
    const wrapper = shallow(<Filter {...props} />);
    wrapper.setState({ showFilterCount: 1 });
    expect(
      wrapper
        .find(FilterButton)
        .at(1)
        .prop('filter'),
    ).toEqual('More');
    expect(wrapper).toMatchSnapshot();
  });
  it('should show picket correctly', () => {
    const wrapper = shallow(<Filter {...props} />);
    wrapper
      .find(FilterButton)
      .at(0)
      .simulate('click', {
        filter: 'color',
        coordinates,
      });
    expect(wrapper.state()).toEqual({
      activeFilters: ['color'],
      coordinates,
      filter: 'color',
      selectedFilters: {},
      showFilterCount: 2,
    });
    expect(wrapper.find(FilterPicker)).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
  it('should clear selected items state after call public clearAll method', function() {
    const wrapper = shallow(<Filter {...props} />);
    wrapper.setState({
      selectedFilters: {
        color: ['1'],
      },
    });
    expect(wrapper.state('selectedFilters')).toEqual({ color: ['1'] });
    wrapper.instance().clearAll();
    expect(wrapper.state('selectedFilters')).toEqual({});
  });
  it('should handle on filter apply correctly', function() {
    const wrapper = shallow(<Filter {...props} />);
    wrapper.setState({
      selectedFilters: {
        size: ['1'],
      },
    });
    expect(wrapper.state('selectedFilters')).toEqual({ size: ['1'] });
    wrapper
      .find(FilterButton)
      .at(0)
      .simulate('click', {
        filter: 'color',
        coordinates,
      });
    expect(wrapper).toMatchSnapshot();
    wrapper
      .find(FilterPicker)
      .first()
      .props()
      .onApply({ color: ['1', '2'] });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.state('selectedFilters')).toEqual({
      size: ['1'],
      color: ['1', '2'],
    });
    expect(wrapper.state('filter')).toEqual(null);
    expect(wrapper.state('coordinates')).toEqual(null);
    expect(wrapper).toMatchSnapshot();
  });
});
