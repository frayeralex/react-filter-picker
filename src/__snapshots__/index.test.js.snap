// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter/> should handle on filter apply correctly 1`] = `
<div>
  <div>
    <FilterButton
      active={true}
      filter="color"
      key="color"
      onClick={[Function]}
      selectedCount={0}
    />
    <FilterButton
      active={true}
      filter="size"
      key="size"
      onClick={[Function]}
      selectedCount={1}
    />
  </div>
  <FilterPicker
    activeFilters={
      Array [
        "color",
      ]
    }
    coordinates={
      Object {
        "bottom": 30,
        "height": 40,
        "left": 10,
        "right": 10,
        "top": 10,
        "width": 100,
      }
    }
    filterData={
      Object {
        "color": Array [
          Object {
            "id": "1",
            "title": "first",
          },
          Object {
            "id": "2",
            "title": "second",
          },
        ],
        "size": Array [
          Object {
            "id": "1",
            "title": "first",
          },
          Object {
            "id": "2",
            "title": "second",
          },
        ],
      }
    }
    md={
      MobileDetect {
        "_cache": Object {},
        "maxPhoneWidth": 600,
        "ua": "Node.js (darwin; U; rv:v12.6.0) AppleWebKit/537.36 (KHTML, like Gecko)",
      }
    }
    onApply={[Function]}
    selectedFilters={
      Object {
        "size": Array [
          "1",
        ],
      }
    }
  />
</div>
`;

exports[`<Filter/> should handle on filter apply correctly 2`] = `
<div>
  <div>
    <FilterButton
      active={true}
      filter="color"
      key="color"
      onClick={[Function]}
      selectedCount={2}
    />
    <FilterButton
      active={true}
      filter="size"
      key="size"
      onClick={[Function]}
      selectedCount={1}
    />
  </div>
</div>
`;

exports[`<Filter/> should handle on filter apply correctly 3`] = `
<div>
  <div>
    <FilterButton
      active={true}
      filter="color"
      key="color"
      onClick={[Function]}
      selectedCount={2}
    />
    <FilterButton
      active={true}
      filter="size"
      key="size"
      onClick={[Function]}
      selectedCount={1}
    />
  </div>
</div>
`;

exports[`<Filter/> should render with menu include More Filter button 1`] = `
<div>
  <div>
    <FilterButton
      active={false}
      filter="color"
      key="color"
      onClick={[Function]}
      selectedCount={0}
    />
    <FilterButton
      active={false}
      filter="More"
      onClick={[Function]}
      selectedCount={0}
    />
  </div>
</div>
`;

exports[`<Filter/> should render without crashing 1`] = `
<div>
  <div>
    <FilterButton
      active={false}
      filter="color"
      key="color"
      onClick={[Function]}
      selectedCount={0}
    />
    <FilterButton
      active={false}
      filter="size"
      key="size"
      onClick={[Function]}
      selectedCount={0}
    />
  </div>
</div>
`;

exports[`<Filter/> should show picket correctly 1`] = `
<div>
  <div>
    <FilterButton
      active={true}
      filter="color"
      key="color"
      onClick={[Function]}
      selectedCount={0}
    />
    <FilterButton
      active={false}
      filter="size"
      key="size"
      onClick={[Function]}
      selectedCount={0}
    />
  </div>
  <FilterPicker
    activeFilters={
      Array [
        "color",
      ]
    }
    coordinates={
      Object {
        "bottom": 30,
        "height": 40,
        "left": 10,
        "right": 10,
        "top": 10,
        "width": 100,
      }
    }
    filterData={
      Object {
        "color": Array [
          Object {
            "id": "1",
            "title": "first",
          },
          Object {
            "id": "2",
            "title": "second",
          },
        ],
        "size": Array [
          Object {
            "id": "1",
            "title": "first",
          },
          Object {
            "id": "2",
            "title": "second",
          },
        ],
      }
    }
    md={
      MobileDetect {
        "_cache": Object {},
        "maxPhoneWidth": 600,
        "ua": "Node.js (darwin; U; rv:v12.6.0) AppleWebKit/537.36 (KHTML, like Gecko)",
      }
    }
    onApply={[Function]}
    selectedFilters={Object {}}
  />
</div>
`;
